clients/include/testing_spmv.hpp

```diff
@@ -0,0 +1,246 @@
+/* ************************************************************************
+ * Copyright 2018-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+
+#include "cblas_interface.hpp"
+#include "flops.hpp"
+#include "norm.hpp"
+#include "rocblas.hpp"
+#include "rocblas_init.hpp"
+#include "rocblas_math.hpp"
+#include "rocblas_random.hpp"
+#include "rocblas_test.hpp"
+#include "rocblas_vector.hpp"
+#include "unit.hpp"
+#include "utility.hpp"
+
+template <typename T>
+void testing_spmv_bad_arg()
+{
+    rocblas_fill         uplo  = rocblas_fill_upper;
+    rocblas_int          N     = 100;
+    rocblas_int          incx  = 1;
+    rocblas_int          incy  = 1;
+    T                    alpha = 0.6;
+    T                    beta  = 0.6;
+    rocblas_local_handle handle;
+
+    size_t abs_incx = incx >= 0 ? incx : -incx;
+    size_t abs_incy = incy >= 0 ? incy : -incy;
+    size_t size_A   = N * N;
+    size_t size_x   = N * abs_incx;
+    size_t size_y   = N * abs_incy;
+
+    // allocate memory on device
+    device_vector<T> dA(size_A);
+    device_vector<T> dx(size_x);
+    device_vector<T> dy(size_y);
+    if(!dA || !dx || !dy)
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    EXPECT_ROCBLAS_STATUS(rocblas_spmv<T>(nullptr, uplo, N, &alpha, dA, dx, incx, &beta, dy, incy),
+                          rocblas_status_invalid_handle);
+
+    EXPECT_ROCBLAS_STATUS(rocblas_spmv<T>(handle, uplo, N, nullptr, dA, dx, incx, &beta, dy, incy),
+                          rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(
+        rocblas_spmv<T>(handle, uplo, N, &alpha, nullptr, dx, incx, &beta, dy, incy),
+        rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(
+        rocblas_spmv<T>(handle, uplo, N, &alpha, dA, nullptr, incx, &beta, dy, incy),
+        rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(rocblas_spmv<T>(handle, uplo, N, &alpha, dA, dx, incx, nullptr, dy, incy),
+                          rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(
+        rocblas_spmv<T>(handle, uplo, N, &alpha, dA, dx, incx, &beta, nullptr, incy),
+        rocblas_status_invalid_pointer);
+}
+
+template <typename T>
+void testing_spmv(const Arguments& arg)
+{
+    rocblas_int N    = arg.N;
+    rocblas_int incx = arg.incx;
+    rocblas_int incy = arg.incy;
+
+    host_vector<T> alpha(1);
+    host_vector<T> beta(1);
+    alpha[0] = arg.alpha;
+    beta[0]  = arg.beta;
+
+    rocblas_fill uplo = char2rocblas_fill(arg.uplo);
+
+    size_t abs_incx = incx >= 0 ? incx : -incx;
+    size_t abs_incy = incy >= 0 ? incy : -incy;
+
+    size_t size_A = size_t(N) * N;
```

Comment:
The size of A can be `(n * (n + 1) / 1)`.

Reply:
Don't you mean tri_count :-)    Will reduce.